Products/ZenModel/migrate/addIndexesForDevice.py

```diff
@@ -0,0 +1,38 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2017, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import Migrate
+from Products.Zuul.catalog.model_catalog_init import reindex_model_catalog
+from Products.Zuul.catalog.indexable import DeviceIndexable
+from Products.ZenModel.ZMigrateVersion import SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION
+
+class AddIndexesForDevice(Migrate.Step):
+    """
+    Adds new indexes for devices in scope of adding new router which gives a possibility
+    to pull data directly from SOLR without touching ZODB.
+    """
+
+    version = Migrate.Version(SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION)
+
+    def cutover(self, dmd):
+
```

Comment:
Please add some logging to indicate that all devices are going to be reindexed since it is an operation that can take a while

Reply:
Also, before reindexing, check that this migrate has not already been run so it only runs once. Fields like `collector` or `pythonClass` should be in all devices, so they are good candidates to run a test query.

Reply:
done